cedar-policy-generators/src/hierarchy.rs

```diff
@@ -5,6 +5,41 @@ use arbitrary::{Arbitrary, Unstructured};
 use cedar_policy_core::ast::{Eid, Entity, EntityUID, Name};
 use cedar_policy_core::entities::{Entities, TCComputation};
 
+// EntityUIDs with the mappings to their indices in the container
+// This is used to generate an entity that is lexicographically smaller/greater than the input entity
+#[derive(Debug, Clone)]
+struct EntityUIDs {
+    pub indices: HashMap<EntityUID, usize>,
+    pub uids: Vec<EntityUID>,
+}
+
+impl From<Vec<EntityUID>> for EntityUIDs {
+    fn from(value: Vec<EntityUID>) -> Self {
+        Self {
+            indices: (0..value.len())
+                .into_iter()
+                .map(|idx| (value[idx].clone(), idx))
+                .collect(),
+            uids: value,
+        }
+    }
+}
+
+impl<'a> From<&'a EntityUIDs> for &'a [EntityUID] {
+    fn from(value: &'a EntityUIDs) -> Self {
+        &value.uids
+    }
+}
+
+impl EntityUIDs {
+    // Get an `EntityUID` whose index is greater than `uid`'s
```

Comment:
This comment implies we're getting a single `EntityUID`, but the function actually returns a slice of `EntityUID`s